packages/peregrine/lib/talons/Checkout/useOverview.js

```diff
@@ -38,10 +38,13 @@ export const useOverview = props => {
         submitOrder();
     }, [submitOrder]);
 
-    const currencyCode =
-        (cart && cart.totals && cart.totals.quote_currency_code) || 'USD';
-    const numItems = (cart && cart.details && cart.details.items_qty) || 0;
-    const subtotal = (cart && cart.totals && cart.totals.subtotal) || 0;
+    const { details } = cart;
+    // TODO: Cart state is wiped and this component re-renders and chokes without these
+    // defaults. Is there a better way to handle this?
+    const { items = [], prices = {} } = details;
+    const { grand_total: grandTotal = {} } = prices;
```

Comment:
There is also a discrepancy between grand totals on submit order vs grand total in the gql response for cart details. I wonder if this is present in `develop`. I think we'd have to update the cart total _after_ selecting a shipping method. #2018 _may_ resolve this, otherwise we could hold off until we work on the new UI/flow.

[![Image from Gyazo](https://i.gyazo.com/0f53d76e84bceb3ce8f829fa1e8d6b35.png)](https://gyazo.com/0f53d76e84bceb3ce8f829fa1e8d6b35)

Reply:
If we'd prefer to keep the behavior consistent with `develop`, we could use `subtotal_including_tax` instead of `grand_total`. It does sound best to defer this to a future re-factor that starts displaying taxes and shipping, but easy change if you want.